atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,262 @@
+
+        component = get_preferred_label(labels, 'com.redhat.component')
+        version = get_preferred_label(labels, 'version')
+        release = get_preferred_label(labels, 'release')
+
+        source = self.workflow.source
+        if not isinstance(source, GitSource):
+            raise RuntimeError('git source required')
+
+        extra = {'image': {'autorebuild': is_rebuild(self.workflow)}}
+        koji_task_id = metadata.get('labels', {}).get('koji-task-id')
+        if koji_task_id is not None:
+            self.log.info("build configuration created by Koji Task ID %s",
+                          koji_task_id)
+            try:
+                extra['container_koji_task_id'] = int(koji_task_id)
+            except ValueError:
+                self.log.error("invalid task ID %r", koji_task_id, exc_info=1)
+
+        fs_result = self.workflow.prebuild_results.get(AddFilesystemPlugin.key)
+        if fs_result is not None:
+            try:
+                fs_task_id = fs_result['filesystem-koji-task-id']
+            except KeyError:
+                self.log.error("%s: expected filesystem-koji-task-id in result",
+                               AddFilesystemPlugin.key)
+            else:
+                try:
+                    task_id = int(fs_task_id)
+                except ValueError:
+                    self.log.error("invalid task ID %r", fs_task_id, exc_info=1)
+                else:
+                    extra['filesystem_koji_task_id'] = task_id
+
+        help_result = self.workflow.prebuild_results.get(AddHelpPlugin.key)
+        if isinstance(help_result, dict) and 'help_file' in help_result and 'status' in help_result:
+            if help_result['status'] == AddHelpPlugin.NO_HELP_FILE_FOUND:
+                extra['image']['help'] = None
+            elif help_result['status'] == AddHelpPlugin.HELP_GENERATED:
+                extra['image']['help'] = help_result['help_file']
+            else:
+                self.log.error("Unknown result from add_help plugin: %s", help_result)
+
+        build = {
+            'name': component,
+            'version': version,
+            'release': release,
+            'source': "{0}#{1}".format(source.uri, source.commit_id),
+            'start_time': start_time,
+            'end_time': int(time.time()),
+            'extra': extra,
+        }
+
+        if self.metadata_only:
+            build['metadata_only'] = True
+
+        return build
+
+    def combine_metadata_fragments(self):
+        try:
+            metadata = get_build_json()["metadata"]
+            self.build_id = metadata["name"]
+        except KeyError:
+            self.log.error("No build metadata")
+            raise
+
+        metadata_version = 0
+        build = self.get_build(metadata)
+
+        metadatas = self.workflow.postbuild_results.get(FetchWorkerMetadataPlugin.key)
+        buildroot = self.get_buildroot(metadatas)
+        output = self.get_output(metadatas)
```

Comment:
Very minor suggestion: 'buildroot' and 'output' could both be named as plurals, like 'buildroots = self.get_buildroots(metadatas)'.

It's really confusing and unfortunate that the CG metadata format requires 'buildroots' (correct, makes sense) but 'output' (which ought to be called 'outputs' as the value is a list).

Anyway, it's fine as-is, just might be easier to read if the only 'huh?' thing is `'output': outputs,`, which is the fault of the CG metadata format spec.